src/EventSubscriber/PageHeaderSubscriber.php

```diff
@@ -14,6 +14,20 @@
  */
 class PageHeaderSubscriber implements EventSubscriberInterface {
 
+  /**
+   * The config.factory service.
+   *
+   * @var Drupal\Core\Config\ImmutableConfig
+   */
+  protected $settings;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($config_factory) {
+    $this->settings = $config_factory->get('localgov_guides.settings');
```

Comment:
Getting the settings from the config factory in the constructor is [considered an anit-pattern](https://mglaman.dev/blog/dependency-injection-anti-patterns-drupal#:~:text=Assigning%20a%20configuration%20object%20to%20a%20property%20instead%20of%20the%20configuration%20factory). Instead get the config factory in the constructor and then get the settings when needed. We ran into an interesting issue with this at BHCC with a [contrib module causing some strange side effects](https://www.drupal.org/project/msqrole/issues/3392181).

Reply:
Good to know. Fixed.